controller/control.go

```diff
@@ -831,6 +831,54 @@ func (c *Controller) WriteAt(b []byte, off int64) (int, error) {
 	return n, err
 }
 
+func (c *Controller) Sync() (err error) {
+	c.Lock()
+	if c.ReadOnly == true {
+		err := fmt.Errorf("Mode: ReadOnly")
+		c.Unlock()
+		time.Sleep(1 * time.Second)
+		return err
+	}
+	defer c.Unlock()
+	err = c.backend.Sync()
+	if err != nil {
+		errh := c.handleErrorNoLock(err)
+		if bErr, ok := err.(*BackendError); ok {
+			if len(bErr.Errors) > 0 {
+				for address := range bErr.Errors {
+					c.RemoveReplicaNoLock(address)
+				}
+			}
+		}
+		return errh
```

Comment:
here, this may return nil when c.backend.Sync returned errors with more than (m.writers/2), but, a replica being in rw, errh becomes nil. Need to handle this case.

Reply:
same can apply for unmap as well